registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/ClassSweepCompilerPluginComponent.scala

```diff
@@ -0,0 +1,42 @@
+package org.virtuslab.ash
+
+import org.virtuslab.ash.RegistrationCheckerCompilerPlugin.classSweepPhaseName
+
+import scala.collection.mutable
+import scala.tools.nsc.{Global, Phase}
+import scala.tools.nsc.plugins.PluginComponent
+
+class ClassSweepCompilerPluginComponent(options: RegistrationCheckerOptions, override val global: Global)
+    extends PluginComponent {
+  import global._
+  override val phaseName: String = classSweepPhaseName
+  override val runsAfter: List[String] = List("refchecks")
+  override def description: String = s"searches for direct descendants of classes annotated with serializability trait"
```

Comment:
Hmmm why do we care about _direct descendants_? Don't we check just for the annotated classes itself, whether they're referenced from the serializer class?

Reply:
That way the only class that is annotated, is the serializability trait itself.
They must be direct descendants, because we need to know how to serialize all named classes of serializability trait.